### ISSUE TITLE
wiki page "Using libGDX" is a little bit outdated

### ISSUE THREAD

**Issue by LionHere** [kotlin-graphics/imgui#62]
[Using libGDX](https://github.com/kotlin-graphics/imgui/wiki/Using-libGDX) is using some API that are deprecated, so I wrote another one (in Java, cause I don't know Kotlin 😢).

[Using libGDX (in Java)](https://github.com/LionHere/kotlin-graphics-imgui-wiki-pr/wiki/Using-libGDX-(in-Java))

But I didn't cover the "libGDX texture" or "Input callbacks" part, so this page should (at most) be at the same hierarchy level with [Using libGDX](https://github.com/kotlin-graphics/imgui/wiki/Using-libGDX).

I don't know if this is appropriate to fire a new issue, so feel free to close it. And feel free to pull it into the wiki. 😄

**Comment by klianc09** [comment]
I wanted to use imgui together with (Java) LibGDX for a while now, but never got around to it. This is a great excuse to finally get to it.
I should be able to look into your code next week and will add it to the wiki if it makes sense and/or is useful.

**Comment by elect86** [comment]
Hi LionHere,

that's great, I'll add you to the collaborators so that you can add that page yourself :)

**Comment by LionHere** [comment]
> Hi LionHere,
> 
> that's great, I'll add you to the collaborators so that you can add that page yourself :)

If that if okay with you, then great! I'll have time to do this in a couple of days!

**Comment by klianc09** [comment]
Okay, so I updated my example project and I noticed a few things (some not directly related to this issue I guess):
1. @LionHere I like the structure of your wiki page better than the current. We should certainly add it to the wiki. The current Kotlin example is out-dated and should be updated or removed, but I can't do Kotlin, so I can only remove it. I suggest replacing the Kotlin example completely with your Java example, until someone adds an up-to-date kotlin example again. I guess the code is almost the same for both anyways.
2. The gradle code to get the lwjgl native dependencies from [Install](https://github.com/kotlin-graphics/imgui/wiki/Install) won't work for "lwjgl-vulkan" since only macos natives exist for it [(see this)](https://github.com/LWJGL/lwjgl3/issues/412). Therefore the build fails using it on windows/linux, I fixed this in my project by [adding a check for that dependency](https://github.com/klianc09/imguiexample/blob/b42b455dadda374fe9b6c76d673e0839e13fd0fe/core/build.gradle), and would adapt it on the Install wiki page, if there is no objection.
3. My project runs fine also without the natives. Why do we need to explicitly add them in the build script in the first place? Does it maybe only work, because LibGDX already provides them, or am I missing something obvious?

**Comment by LionHere** [comment]
> My project runs fine also without the natives. Why do we need to explicitly add them in the build script in the first place? Does it maybe only work, because LibGDX already provides them, or am I missing something obvious?

So I did a little testing... It seemed that when I was using [czyzby/gdx-setup](https://github.com/czyzby/gdx-setup), I actually did't need to add any dependency explicitly for LWJGL, and Gradle would take care for it when I add imgui dependency.

But when I was using the official gdx-setup, not adding LWJGL dependency will casue some runtime exception relating to dependency.

I don't have an actual understanding of how this works, and maybe it is because LibGDX already provides some of the dependencies.  I am going to update part of the [Using libGDX](https://github.com/kotlin-graphics/imgui/wiki/Using-libGDX) page, and let someone else fill in the blanks relating to the dependency problem.
